storage/inmemory/src/lib.rs

```diff
@@ -13,316 +13,61 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use common::sync;
-use std::collections::BTreeMap;
-use storage_core::{
-    schema::{self, Schema},
-    traits, transaction, Data,
-};
-
-// These store the data
-type StoreMapSingle = BTreeMap<Data, Data>;
-type StoreMapMulti = BTreeMap<Data, Vec<Data>>;
-
-pub enum StoreMap {
-    Single(StoreMapSingle),
-    Multi(StoreMapMulti),
-}
-
-// Set of store maps, one per db index.
-type StoreMapSet = BTreeMap<&'static str, StoreMap>;
-
-// These store changes to the data that could be commited or discarded
-type DeltaMapSingle = BTreeMap<Data, Option<Data>>;
-
-// Vector delta
-#[allow(unused)]
-enum VecDelta<T> {
-    Set(Vec<T>),
-    Modify { add: Vec<T>, del: Vec<T> },
-}
-
-type DeltaMapMulti = BTreeMap<Data, VecDelta<Data>>;
-
-enum DeltaMap {
-    Single(DeltaMapSingle),
-    Multi(DeltaMapMulti),
-}
+use storage_core::{adaptor, backend, Data, DbDesc, DbIndex, Result as BackendResult};
```

Comment:
I don't think `Result as BackendResult` aliasing is necessary.

Reply:
Will have a look. I'd like to avoid unqualified `Result` since I find it can be easily mistaken for `std::result::Result` by someone not familiar with the pattern of defining `Result<T, E>` specialized for some fixed `E`.

Reply:
It's quite a common pattern in rust to see Result being an alias for something else. Don't anyhow and its users do that quite often?

Reply:
> It's quite a common pattern in rust to see Result being an alias for something else. Don't anyhow and its users do that quite often?

They do. However, I like to refer to them using a qualified path, like `anyhow::Result` rather than importing it (shadowing `Result` from `std`) and using it as just `Result`. WIll do something similar here, would that be okay?

Reply:
Yeah, sure. That's fine.

Reply:
Personally I prefer to shadow `std::result::Result` only if no other result-types are used in the crate/module. It is usually the case with `anyhow`. Additionally I think it is better for the crate (`storage_core` in this case) to export a result with some more specific name. Otherwise every user of a crate will be forced to introduce an alias himself or use the qualified name everywhere.